client/my-sites/site-settings/podcasting-details/index.jsx

```diff
@@ -342,6 +354,17 @@ class PodcastingDetails extends Component {
 
 		const fieldsToUpdate = { podcasting_category_id: String( category.ID ) };
 
+		if ( category.ID === settings.podcasting_category_id ) {
```

Comment:
This isn't necessarily specific to this PR, but it is not clear to me what the difference between `settings` and `fields` is, and why we have to check both in this method. 

It is also not clear to me why `state` is used here, instead handling things in the same flow as other props are. For example, we do not handle enabling/disabling the rest of the form through state. So, why handle showing this notice, which also depends on the category changing, with `state`?

Reply:
>This isn't necessarily specific to this PR, but it is not clear to me what the difference between settings and fields is, and why we have to check both in this method.

`settings` are the saved values returned from the API, whereas `fields` are the current values of the form field (can be different if the form hasn't been saved).

>It is also not clear to me why state is used here, instead handling things in the same flow as other props are. For example, we do not handle enabling/disabling the rest of the form through state. So, why handle showing this notice, which also depends on the category changing, with state?

Great question that I don't have an answer for :). I've updated the PR to used connected props to follow suit with the rest of the component.